packages/sdk/src/stores/google-datastore/store.ts

```diff
@@ -22,37 +25,61 @@ export default function <D>(kind: string, store: Datastore): SortedStore<string,
     });
   }
   async function has(id: string) {
-    try {
-      await get(id);
-      return true;
-    } catch (err) {
-      return false;
-    }
+    // horribly ineficient to actually query data to see if it exists, but cannot find a better way
+    return exists(await get(id));
   }
   async function del(id: string) {
     await store.delete(makeKey(id));
   }
   async function entries() {
-    return [];
+    const [results] = await store.createQuery(kind).run();
+    return results.map((result) => {
+      return [result[store.KEY].name, result];
+    }) as [string, D][];
   }
   async function values() {
-    return [];
+    const [results] = await store.createQuery(kind).run();
+    return results;
   }
+  // all this does is map over full entries to return the id of the entry. have not found a better way to query this.
   async function keys() {
-    return [];
+    const [results] = await store.createQuery(kind).run();
+    return results.map((result) => {
+      return result[store.KEY].name;
+    }) as string[];
   }
+  // theres no way to really do built into the store client. Google recommends managing a size entry yourself.
   async function size() {
-    return 0;
+    throw new Error("size not supported in google store");
+    // typescript complains if theres no return value
+    return NaN;
   }
-  // TODO: implement these functions
+  // this actually queries all values, then batches them to delete N at a time. This is a horrible way
+  // to do this, but I could not find a better way to "drop" all kinds from the table.
   async function clear() {
-    throw new Error("clear is not implemented");
+    await highland(store.createQuery(kind).runStream())
+      .map((val: any) => {
+        return val[store.KEY];
+      })
+      .batch(500)
```

Comment:
Why'd you pick 500? Are we sure we need to do this? Will the library handle this sort of thing for us?

Reply:
we dont actually need this particular function in any production apps. i scoured google for a way to to clear tables or "kinds" and probably saw references to numbers in the ballpark of 500-1000 batches when deleting things. as far as i know theres no other way to do this, which seems really crazy but it was a theme when working with datastore. really simple operations i would take for granted either dont exist or are really inefficient.  

i would be happy to remove this since its not strictly needed, but its just useful for tests when clearing data.  also would be happy if theres a better way to do this

Reply:
heres a thread that talks about it: https://groups.google.com/g/google-appengine-go/c/eOu1dDJ7MNg

Reply:
Okay, then that seems fine to leave at 500, although this is something I'm surprised that the sdk/client library doesn't handle this for us.